### ISSUE TITLE
Peer Review - Final Report

### ISSUE THREAD

**Issue by Wang-Siyuan** [AaronZang/ORIE4741-Home-Purchase-Assistance#8]
Things I liked:
- Spent a lot of effort in understanding the feature and selecting features that are essential to the problem. I also appreciate the fact that you listed out all the transformation you have done to the input data. (although not sure if "dummy coding" means one-hot encoding)
- I really liked how you computed the average price for each neighborhood and replaced the neighborhood feature data with the precomputed value.  That's an effective way of incorporating the information without creating too many new columns.
- Demonstrated many attempts in fitting different model that we learned in class and use validation data set to figure out the best model to use. 

Things that could be improved:
- In the report, you mentioned that you choose 8 values for your building class category features and eliminated the data that doesn't fall into these 8 categories.  This seemed a bit arbitrary to me.  What's the data size after the data filtering? Why are these 8 categories important? If we were using this software in a production environment, we cannot simply choose certain feature nominal values to be included in our model.
- You didn't demonstrate how your model would scale when the data size change. You have done some cross-validation using a consistent data size. It would be interesting to see if how the model performs with a smaller data set.
- After trying an array of models and narrow down to one loss function and regularizer that seem to performs the best, there wasn't much effort in improving the model by looking at the result from the first time.  There are other things you can try such as changing your regularizer coefficient(using k-fold cross validation), or standardizing your input data set, etc.  It would be great if you could take iterative approach to further improve your model.
- The report mentioned that you have used proximal gradient method, but I could not find that in your code.

**Comment by yw762** [comment]
we consider only 8 types of building categories because only these are residential homes. Other types such as "factories" and "hospitals" are clearly outliers and are not within the scope of this project.

**Comment by AaronZang** [comment]
@Wang-Siyuan , thanks for the comment.

- Dummy coding means "one-hot encoding".
- Most (90%+) of the records in our dataset belong to those 8 categories we chose. Since residential housing and commercial housing may have different patterns, we felt like it's reasonable to only build the model for residential housing.
- We did some experiments on smaller dataset: if we use only 10% of the data for training, the results are almost the same. If we use <%1 of the data for training, the models would have an overfitting issue.
- We'll upload our code to the repo.

Hope they clarify your concerns!
